### PR TITLE
suggestFilename: use modeService instead of mime

### PR DESCRIPTION
**Pull request by aeschli**
This PR is for #102032

#102101 will change that modeService.getExtension will first return the file extension that come from the 'primary' language contribution. The 'primary' language contribution is the one that defines the language configuration.

This PR simplifies the code for suggestFilename to not use the mime registry (which sorts mime types by the order that they were registered), but using the modeService.

### CODE REVIEW

**Comment by bpasero** [comment]
Let me review, thanks.

**Review by bpasero — APPROVED** [review]
Looks like this needs other changes to work, so I would not merge this without the other PR as it currently is not working.

**Comment by bpasero** [comment]
@aeschli once this landed I think we should do a pass over all our languages and ensure that the extension is first that makes most sense. E.g. for JSONC I see `.hintrc` seems to be first but `.jsonc` should be put first.

**Comment by aeschli** [comment]
Thanks @bpasero. I made a pass over all languages and pushed https://github.com/microsoft/vscode/commit/4c99db1a05687a67c1d214361946b18e5f65e0cf

**Pull request merged**
